Archipelago.MultiClient.Net.Analyzers/Archipelago.MultiClient.Net.Analyzers.Test/ItemFlagComparisonsTest.cs

```diff
@@ -0,0 +1,427 @@
+            }
+        }
+    }
+}";
+            DiagnosticResult[] expected = [
+                VerifyCS.Diagnostic("MULTICLIENT003").WithLocation(0),
+                VerifyCS.Diagnostic("MULTICLIENT003").WithLocation(1)
+            ];
+            await VerifyCS.VerifyAnalyzerAsync(test, expected);
+        }
+
+        [TestMethod]
+        public async Task VerifyComparisonToNoneDoesNotYieldDiagnostic()
+        {
+            string test = @"
+using System;
+using Archipelago.MultiClient.Net.Enums;
+
+namespace MyClient
+{
+    class MyClass
+    {
+        public bool Test()
+        {
+            ItemFlags i = ItemFlags.Advancement;
+            switch (i)
+            {
+                {|#0:case ItemFlags.Advancement:|}
+                    return true;
+                case ItemFlags.None:
+                    return false;
+                default:
+                    return false;
+            }
+        }
+    }
+}";
+            DiagnosticResult expected = VerifyCS.Diagnostic("MULTICLIENT003").WithLocation(0);
+            await VerifyCS.VerifyAnalyzerAsync(test, expected);
+        }
+
+        [TestMethod]
+        public async Task VerifyFallthroughItemFlagsInSwitchStatementYieldsDiagnostics()
+        {
+            string test = @"
+using System;
+using Archipelago.MultiClient.Net.Enums;
+
+namespace MyClient
+{
+    class MyClass
+    {
+        public bool Test()
+        {
+            ItemFlags i = ItemFlags.Advancement;
+            switch (i)
+            {
+                {|#0:case ItemFlags.Advancement:|}
+                {|#1:case ItemFlags.Trap:|}
+                    return false;
+                default:
+                    return false;
+            }
+        }
+    }
+}";
+            DiagnosticResult[] expected = [
+                VerifyCS.Diagnostic("MULTICLIENT003").WithLocation(0),
+                VerifyCS.Diagnostic("MULTICLIENT003").WithLocation(1)
+            ];
+            await VerifyCS.VerifyAnalyzerAsync(test, expected);
+        }
+
+        [TestMethod]
+        public async Task VerifyMixedUseItemFlagsInSwitchStatementYieldsDiagnostic()
+        {
+            string test = @"
+using System;
+using Archipelago.MultiClient.Net.Enums;
+
+namespace MyClient
+{
+    class MyClass
+    {
+        public bool Test()
+        {
+            ItemFlags i = ItemFlags.Advancement;
+            switch (i)
+            {
+                case var f when f.HasFlag(ItemFlags.Advancement):
+                    return true;
+                {|#0:case ItemFlags.Trap:|}
+                    return true;
+                default:
+                    return false;
+            }
+        }
+    }
+}";
+            DiagnosticResult expected = VerifyCS.Diagnostic("MULTICLIENT003").WithLocation(0);
+            await VerifyCS.VerifyAnalyzerAsync(test, expected);
+        }
+    
+
+        [TestMethod]
+        public async Task VerifyItemFlagsInSwitchExpressionYieldsNoDiagnostic()
```

Comment:
assuming that we were to keep switch expressions out of this, this is not a good test as none of the cases here would violate the rule normally.

Reply:
That test may not be named in the best way, but its intent is to ensure that no diagnostic is issued for a compliant case.

Reply:
I guess I'm saying if we are leaving switch expressions intentionally out of scope we should also have a test which asserts no diagnostic on a switch expression which would be a violation (with a comment that it should be changed if/when we do support switch expressions)

Reply:
I've updated the test case to include a problematic match in the switch expression, while still asserting it yields no diagnostic for switch expressions. 

Sorry for not tackling switch expressions, but I'm nearing completion on this and it'd take me another 2 months to gather the spirit bomb energy to do more. 😓